client/src/scenes/Battle.ts

```diff
@@ -7,26 +7,66 @@ export default class Battle extends Phaser.Scene {
     private client: Colyseus.Client
     private cursors!: Phaser.Types.Input.Keyboard.CursorKeys //trust that this will exist with the !
     private faune!: Phaser.Physics.Arcade.Sprite
+    private monsters: Phaser.Physics.Arcade.Sprite[] = [];
     private playerEntities: { [sessionId: string]: any } = {};
     private room!: Colyseus.Room
+    currentRound: number = 1;
+    roundText!: Phaser.GameObjects.Text;
+    private timerText!: Phaser.GameObjects.Text;
+    private ROUND_DURATION_MINUTES = 1;
+    MINUTE_TO_MILLISECONDS: number = 60 * 1000;
+    private remainingTime: number = this.ROUND_DURATION_MINUTES * 60;
+
     inputPayload = {
         left: false,
         right: false,
         up: false,
         down: false,
     };
 
+
     constructor() {
         super('battle')
         this.client = new Colyseus.Client('ws://localhost:2567');
     }
 
     preload() {
         //create arrow and spacebar
-        this.cursors = this.input.keyboard.createCursorKeys()
+
+    }
+
+    //Method to spawn monsters at random positions.
+    //Called when the scene is created and when a new round starts
+    private spawnMonsters() {
+        // Remove existing monsters
+        this.monsters.forEach(monster => monster.destroy());
+        this.monsters = [];
+
+        // Spawn new monsters
+        for (let i = 0; i < 5; i++) { // Change this number to spawn more or less monsters
+            const x = Phaser.Math.Between(50, 750); // Change these numbers to the size of your game world
+            const y = Phaser.Math.Between(50, 550); // Change these numbers to the size of your game world
+            const monster = this.physics.add.sprite(x, y, 'dragon', 'free_4x_0');
+            monster.setScale(0.5); // Scales the sprite to 50% of its original size
+            this.monsters.push(monster);
+        }
+    }
+
+    private setUpTimer() {
+        this.timerText = this.add.text(0, 100, `Time: ${this.remainingTime}`, { fontSize: '15px', fill: '#fff' });
+
+        // Create a Phaser Timer Event that updates the timer text every second
+        this.time.addEvent({
+            delay: 1000, // 1000 milliseconds = 1 second
+            callback: this.updateTimer,
+            callbackScope: this,
+            loop: true
+        });
     }
 
     async create() {
+        this.cursors = this.input.keyboard.createCursorKeys()
```

Comment:
Lets standardise to put those that can be loaded, like cursor/keyboard input into the preload?